js/errors.js

```diff
@@ -17,19 +18,35 @@
     module.IntegrityConflictError = IntegrityConflictError;
     module.DuplicateConflictError = DuplicateConflictError;
 
+    /**
+     * @memberof ERMrest
+     * @param  {int} code           http error code
+     * @param  {string} status      message status/title in the modal box
+     * @param  {string} message     main user error message
+     * @param  {string} subMessage  technical details about the error. Appear in collapsible span in the modal box
+     * @constructor
+     */
+    function ErmrestError(code, status, message, subMessage) {
+        this.code = code;
+        this.status = status;
+        this.message = message;
+        this.subMessage = subMessage;
+    }
+
+    ErmrestError.prototype = Object.create(Error.prototype);
+    ErmrestError.prototype.constructor = ErmrestError;
     /**
      * @memberof ERMrest
      * @param {string} status the network error code
      * @param {string} message error message
      * @constructor
      */
     function TimedOutError(status, message) {
-        this.code = 0;
-        this.status = status;
-        this.message = message;
+        var code = 0;
```

Comment:
You don't need to create a variable here. You could just call it with the right code. Now that I am thinking about it, it might be better if you create a object of HTTPErrorCodes.

```
{
  BAD_REQUEST: 400,
  UNAUTHORIZED: 401,
  FORBIDDEN: 403,
 ...,
}
```
For these cases that the http code doesn't make sense just pass 0.

Although I'm not sure why for timeout error we're using 0, we can use it's appropriate code (408). We should also change the switch-case for the timeout to use it's own status code.

Reply:
Don't change the code for timeout yet, let's discuss about this first.